src/cluster/connectionBuilder.js

```diff
@@ -1,11 +1,18 @@
 const Connection = require('../network/connection')
-const { KafkaJSNonRetriableError } = require('../errors')
-const shuffle = require('../utils/shuffle')
+const { KafkaJSConnectionError, KafkaJSNonRetriableError } = require('../errors')
 
-const validateBrokers = brokers => {
-  if (!brokers || brokers.length === 0) {
-    throw new KafkaJSNonRetriableError(`Failed to connect: expected brokers array and got nothing`)
+const getBrokers = async brokers => {
+  const dynamic = typeof brokers === 'function'
+
+  const list = dynamic ? await brokers() : brokers
```

Comment:
Make sure to catch this so that you can wrap the error in a `new KafkaJSConnectionError(error)` if it rejects.

Reply:
yes, updated getBrokers method to be more explicit